internal/runners/mcp/downloadlogs/downloadlogs.go

```diff
@@ -20,29 +23,73 @@ func New(p *primer.Values) *DownloadLogsRunner {
 }
 
 type Params struct {
-	logUrl string
+	LogUrl string
 }
 
-func NewParams(logUrl string) *Params {
-	return &Params{
-		logUrl: logUrl,
-	}
+func NewParams() *Params {
+	return &Params{}
+}
+
+// Example: {"body": {"facility": "INFO", "msg": "..."}, "artifact_id": "...", "timestamp": "2025-08-12T19:23:51.702971", "type": "artifact_progress", "source": "build-wrapper", "pid": 19}
+type LogLine struct {
+	Body struct {
+		Msg string `json:"msg"`
+	} `json:"body"`
 }
 
 func (runner *DownloadLogsRunner) Run(params *Params) error {
-	response, err := http.Get(params.logUrl)
+	response, err := http.Get(params.LogUrl)
 	if err != nil {
 		return fmt.Errorf("error while downloading logs: %v", err)
 	}
 	defer response.Body.Close()
 
-	body, err := io.ReadAll(response.Body)
-	if err != nil {
-		return fmt.Errorf("error reading response body: %v", err)
-	}
-	if response.StatusCode != 200 {
+	if response.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(response.Body)
 		return fmt.Errorf("error fetching logs: status %d, %s", response.StatusCode, body)
 	}
-	runner.output.Print(string(body))
+
+	scanner := bufio.NewScanner(response.Body)
+
+	// Read all lines, parse and only store the text messages
+	var lines []string
+	for scanner.Scan() {
+		line := scanner.Text()
+		var logLine LogLine
+		if err := json.Unmarshal([]byte(line), &logLine); err != nil {
+			continue
+		}
+		lines = append(lines, logLine.Body.Msg)
+	}
+	if err := scanner.Err(); err != nil {
+		return fmt.Errorf("error reading log content: %v", err)
+	}
+
+	// Check what lines contain the keyword "error" and print the previous 10 and next 10 lines
+	printedLines := make(map[int]bool)
+	for i, line := range lines {
+		if strings.Contains(strings.ToLower(line), "error") {
```

Comment:
Should perhaps include some more synonyms here, eg.

fail
critical
panic
exception